cli/silence_import.go

```diff
@@ -0,0 +1,67 @@
+package cli
+
+import (
+	"os"
+	"io/ioutil"
+	"encoding/json"
+	"fmt"
+
+	"github.com/prometheus/alertmanager/types"
+	"github.com/spf13/cobra"
+	"github.com/pkg/errors"
+)
+
+var importCmd = &cobra.Command{
+	Use:   "import [JSON file]",
+	Short: "Import silences",
+	Long: `Import alertmanager silences from JSON file or stdin
+
+  This command can be used to bulk import silences from a JSON file
+  created by query command. For example:
+
+  amtool silence query -o json foo > foo.json
+  amtool silence import foo.json
+
+  JSON data can also come from stdin if no param is specified.
+	`,
+	Args: cobra.MaximumNArgs(1),
+	Run:  CommandWrapper(bulkImport),
+}
+
+func bulkImport(cmd *cobra.Command, args []string) error {
+	input := os.Stdin
+	var err error
+
+	if len(args) == 1 {
+		input, err = os.Open(args[0])
+
+		if err != nil {
+			return err
+		}
+	}
+
+	data, err := ioutil.ReadAll(input)
+	if err != nil {
+		return err
+	}
+
+	var silences []types.Silence
+	err = json.Unmarshal(data, &silences)
+	if err != nil {
+		return errors.Wrap(err, "couldn't unmarshal input data, is it in JSON?")
+	}
+
+	for _, silence := range silences {
+		// reset the ID, otherwise alertmanager API will try to replace an existing silence
+		// this *might* create duplicated silences but that should generally be OK
+		silence.ID = ""
```

Comment:
I think we wan't to avoid duplicate silences if possible. Does the alertmanager API accept posting a new silence with the ID field present?

Reply:
It will accept if the silence ID exists and will replace it with the new one. But if the ID doesn't exist then it will return an error, not found. We can check and only remove ID after receive "not found" error but that will create a lot of extra round trip in the backup / restore use case (as every add will have to check and wait for a not found error first).

Reply:
Yeah, don't think we should have duplicated silences as well. Maybe skip the import if silence already exist?

Reply:
Since this is for disaster recovery/data migration, I'm assuming maintaining the IDs wouldn't be important (and wouldn't expect there to be any pre-existing silences in the new instance of AM).

Reply:
I will add a flag to `force` this, it is need as we are running Alertmanager 0.6 and thus cannot update an existing silences. I will keep the silence ID by default as it appear Alertmanager will only update existing one without creating any new silence in that case.